src/Diagnostics/FxCop/SDL.Analyzer.Common/CSharp/Shared/SyntaxNodeHelper.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SDL.Analyzer.Common
+{
+    public sealed class CSharpSyntaxNodeHelper : SyntaxNodeHelper
+    {
+        private static CSharpSyntaxNodeHelper instance = new CSharpSyntaxNodeHelper();
+
+        public static CSharpSyntaxNodeHelper Default { get { return instance; } }
+
+        private CSharpSyntaxNodeHelper()
+        {
+        }
+
+        public override bool ContainsMethodCall(SyntaxNode node, Func<string, bool> predicate)
+        {
+            if (node == null)
+            {
+                return false;
+            }
+
+            return node.DescendantNodesAndSelf().OfType<InvocationExpressionSyntax>().Any(
+                (InvocationExpressionSyntax child) =>
+                {
+                    return child.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>().Any(name => predicate(name.Identifier.ValueText));
+                });
+        }
+
+        public override IMethodSymbol GetCallerMethodSymbol(SyntaxNode node, SemanticModel semanticModel)
```

Comment:
This helper seems unnecessary really, and also not good for performance. If you are doing any analysis within a code block, you can define a CodeBlockStartedAnalyzer, cache the OwningSymbol from the context parameter argument passed to the start action, and access that in all nested registered actions.